packages/docusaurus-plugin-content-blog/src/feed.ts

```diff
@@ -105,13 +114,19 @@ async function defaultCreateFeedItems({
       } = post;
 
       const content = await readOutputHTMLFile(
-        permalink.replace(siteConfig.baseUrl, ''),
+        permalink.replace(baseUrl, ''),
         outDir,
-        siteConfig.trailingSlash,
+        trailingSlash,
       );
       const $ = cheerioLoad(content);
 
-      const blogPostAbsoluteUrl = normalizeUrl([siteUrl, permalink]);
+      const blogPostAbsoluteUrl = applyTrailingSlash(
```

Comment:
LGTM 👍 

We need a test in `feed.test.ts` covering this, maybe copy/paste `it('has feed item for each post'` and add the site trailing slash config?